qcodes/instrument_drivers/stanford_research/SR830_channels.py

```diff
@@ -0,0 +1,628 @@
+        val_mapping = {1: {'none': 0,
+                           'Aux In 1': 1,
+                           'Aux In 2': 2},
+                       2: {'none': 0,
+                           'Aux In 3': 1,
+                           'Aux In 4': 2}}
+        vals = val_mapping[channel].keys()
+        if ratio not in vals:
+            raise ValueError('{} not in {}'.format(ratio, vals))
+        ratio = val_mapping[channel][ratio]
+        disp_val = int(self.ask('DDEF ? {}'.format(channel)).split(',')[0])
+        self.write('DDEF {}, {}, {}'.format(channel, disp_val, ratio))
+        self._buffer_ready = False
+
+    def _get_ch_display(self, channel):
+        val_mapping = {1: {0: 'X',
+                           1: 'R',
+                           2: 'X Noise',
+                           3: 'Aux In 1',
+                           4: 'Aux In 2'},
+                       2: {0: 'Y',
+                           1: 'Phase',
+                           2: 'Y Noise',
+                           3: 'Aux In 3',
+                           4: 'Aux In 4'}}
+        resp = int(self.ask('DDEF ? {}'.format(channel)).split(',')[0])
+
+        return val_mapping[channel][resp]
+
+    def _set_ch_display(self, channel, disp):
+        val_mapping = {1: {'X': 0,
+                           'R': 1,
+                           'X Noise': 2,
+                           'Aux In 1': 3,
+                           'Aux In 2': 4},
+                       2: {'Y': 0,
+                           'Phase': 1,
+                           'Y Noise': 2,
+                           'Aux In 3': 3,
+                           'Aux In 4': 4}}
+        vals = val_mapping[channel].keys()
+        if disp not in vals:
+            raise ValueError('{} not in {}'.format(disp, vals))
+        disp = val_mapping[channel][disp]
+        # Since ratio AND display are set simultaneously,
+        # we get and then re-set the current ratio value
+        ratio_val = int(self.ask('DDEF ? {}'.format(channel)).split(',')[1])
+        self.write('DDEF {}, {}, {}'.format(channel, disp, ratio_val))
+        self._buffer_ready = False
+
+    def _set_units(self, unit: str) ->None:
+        # TODO:
+        # make a public parameter function that allows to change the units
+        for param in [self.X, self.Y, self.R, self.sensitivity]:
+            param.unit = unit
+
+    def _get_input_config(self, s: str) ->int:
+        """
+        Args:
+            s (str): string representative of an integer
+        """
+        mode = self._N_TO_INPUT_CONFIG[int(s)]
+
+        if mode in ['a', 'a-b']:
+            self.sensitivity.set_validator(self._VOLT_ENUM)
+            self._set_units('V')
+        else:
+            self.sensitivity.set_validator(self._CURR_ENUM)
+            self._set_units('A')
+
+        return mode
+
+    def _set_input_config(self, s: str) ->int:
+        if s in ['a', 'a-b']:
+            self.sensitivity.set_validator(self._VOLT_ENUM)
```

Comment:
`set_validator` is deprecated. We should use `self.sensitivity.vals = ` instead.